Info section

martink-rsa opened the pull request:
Added the following:
1. Two info sections. I've tried multiple ways of laying these out and finally settled with the sections height, instead of margin, being used for layout. This allows for more text to be added if this was ever needed.
2. Removed a couple of container divs so the HTML was cleaner. It is now minimal with only what's needed.

I'm having trouble creating internal anchor links and am still trying to figure out a way for them to work in React.

martink-rsa committed:
Info section

jkettmann commented on src/components/InfoSection/InfoSection.style.js:
```diff
@@ -0,0 +1,20 @@
+  }
+`;
+
+export default InfoSectionWrapper;
```
I would always use a named export for style files for consistency

jkettmann commented on src/components/InfoSection/InfoSection.style.js:
```diff
@@ -0,0 +1,20 @@
+  height: 228px;
+  width: 738px;
+  line-height: 27px;
+  h2 {
```
It's better to create separate components for h2 and p. Otherwise your style are coupled to the DOM structure. When you decide to change the element h2 to a h3 for example or the p to a div for some reason you will break your styling.

jkettmann commented on src/styles/globalStyle.js:
```diff
@@ -19,6 +19,24 @@ const GlobalStyle = createGlobalStyle`
+    color: #0087ff;
+    text-decoration: none;
+  }
+  ul {
```
These styles are specific to the InfoSection, so better create separate styled components for ul and li in InfoSection.style.js

jkettmann commented on src/styles/globalStyle.js:
```diff
@@ -19,6 +19,24 @@ const GlobalStyle = createGlobalStyle`
+        font-weight: bold;
+        display: inline-block;
+        width: 9px;
+        margin-left: -40px;
```
Maybe rather set padding-inline-start on the ul element instead of negative margin here if you haven't tried yet

jkettmann commented on src/components/InfoSection/InfoSection.style.js:
```diff
@@ -0,0 +1,20 @@
+const InfoSectionWrapper = styled.section`
+  display: flex;
+  flex-direction: column;
+  height: 228px;
```
Hard-coding the height here is counter-productive when you consider that there might be another info section element added later on. See my comment in the component below

jkettmann commented on src/pages/HomePage/HomePage.js:
```diff
@@ -1,14 +1,36 @@
       <Hero />
-      <Info />
-    </HomePageWrapper>
+      <InfoSection name="how-it-works" title="How it works">
```
I would wrap these two InfoSection components into a wrapper. The wrapper handles the margin to the heatmap and footer. Then you don't need to hard-code the height. The InfoSection component should get a margin-top or margin-bottom that handles the distance between the InfoSection elements

jkettmann reviewed (changes requested):
Looks good, but the styling can be a bit more flexible and more future-proof

martink-rsa committed:
Info section

martink-rsa commented:
I've made quite a few changes.

1. I've merged the styling from `InfoSection` into a new component `Info` where I then display everything with styled-components that are individual elements. 

2. I'm using a fixed min-height container for the main info wrapper so the individual sections have space, then using `space-between` so the sections are spread out. Still feel like I'm not handling this properly.

3. I've added media queries for mobile devices so the site displays better on a mobile device.

jkettmann commented on src/components/Footer/Footer.style.js:
```diff
@@ -17,6 +17,12 @@ const FooterItem = styled.div`
 const FooterLeft = styled(FooterItem)`
   padding-left: 76px;
   justify-content: center;
+  @media (max-width: 768px) {
```
Great thinking with the media queries. I'd put the breakpoint 768px in the theme though

jkettmann commented on src/components/Info/Info.style.js:
```diff
@@ -1,13 +1,52 @@
+  justify-content: space-between;
+  min-height: 348px;
+  max-width: 738px;
+  margin: 130px 0 100px 0;
```
I  guess you use the min-height here to the spacing between the info sections, right? This is not optimal. If this was a real-world project somebody might come and want another section added for more information. The ideal would be that you just drop in another InfoSection in the Info component.

You can achieve that by using the space between the "How it works" and "About" sections as margin-top for the InfoSection component (106px). Then the top margin of this InfoWrapper needs to be changed to 24px

jkettmann commented on src/components/Info/Info.style.js:
```diff
@@ -1,13 +1,52 @@
+`;
+
+const InfoHeading = styled.h2`
+  margin: 0 0 13px 0;
```
This margin can be 14px ;-)

jkettmann commented on src/components/Info/Info.style.js:
```diff
@@ -1,13 +1,52 @@
+`;
+
+const InfoP = styled.p`
+  margin: 0;
```
If you like you can set the `letter-spacing: 0.03px;` You'll see that for long strings this is a bit better. Might also be a global style

jkettmann reviewed (changes requested):
1. and 3. are great. I left an alternative approach for 2.

martink-rsa committed:
Info section

martink-rsa commented:
I've made all the suggested changes, including:

1. Changed the margins so the margin-top is on each section. I couldn't seem to get it correct with your measurements. I've used what I found to fit. I'm not sure if this reflects what you were telling me to do.
2. Added letter-spacing as a global style value.

jkettmann reviewed (approved):
Great work!
